epaint/src/tessellator.rs

```diff
@@ -710,9 +715,89 @@ impl Tessellator {
                 }
                 self.tessellate_text(tex_size, text_shape, out);
             }
+            Shape::QuadraticBezier(quadratic_shape) => {
+                self.tessellate_quadratic_bezier(quadratic_shape, out);
+            }
+            Shape::CubicBezier(cubic_shape) => self.tessellate_cubic_bezier(cubic_shape, out),
+        }
+    }
+
+    pub(crate) fn tessellate_quadratic_bezier(
+        &mut self,
+        quadratic_shape: QuadraticBezierShape,
+        out: &mut Mesh,
+    ) {
+        let options = &self.options;
+        let clip_rect = self.clip_rect;
+        
+        if options.coarse_tessellation_culling
+            && !quadratic_shape.bounding_rect().intersects(clip_rect)
+        {
+            return;
         }
+
+        let points = quadratic_shape.flatten(Some(options.bezier_flattern_tolerence));
+        
+        self.scratchpad_path.clear();
+        if quadratic_shape.closed {
+            self.scratchpad_path.add_line_loop(&points);
+        } else {
+            self.scratchpad_path.add_open_points(&points);
+        }
+        if quadratic_shape.fill != Color32::TRANSPARENT {
+            crate::epaint_assert!(
+                quadratic_shape.closed,
+                "You asked to fill a path that is not closed. That makes no sense."
+            );
+            self.scratchpad_path
+                .fill(quadratic_shape.fill, &self.options, out);
+        }
+        let typ = if quadratic_shape.closed {
+            PathType::Closed
+        } else {
+            PathType::Open
+        };
+        self.scratchpad_path
+            .stroke(typ, quadratic_shape.stroke, &self.options, out);
     }
 
+    pub(crate) fn tessellate_cubic_bezier(
+        &mut self,
+        bezier_shape: CubicBezierShape,
+        out: &mut Mesh,
+    ) {
+        let options = &self.options;
+        let clip_rect = self.clip_rect;
+        if options.coarse_tessellation_culling
+            && !bezier_shape.bounding_rect().intersects(clip_rect)
+        {
+            return;
+        }
+
+        let points = bezier_shape.flatten(Some(options.bezier_flattern_tolerence));
+        
+        self.scratchpad_path.clear();
+        if bezier_shape.closed {
+            self.scratchpad_path.add_line_loop(&points);
+        } else {
+            self.scratchpad_path.add_open_points(&points);
+        }
+        if bezier_shape.fill != Color32::TRANSPARENT {
+            crate::epaint_assert!(
+                bezier_shape.closed,
+                "You asked to fill a path that is not closed. That makes no sense."
+            );
+            self.scratchpad_path
+                .fill(bezier_shape.fill, &self.options, out);
+        }
+        let typ = if bezier_shape.closed {
+            PathType::Closed
+        } else {
+            PathType::Open
+        };
+        self.scratchpad_path
+            .stroke(typ, bezier_shape.stroke, &self.options, out);
+    }
```

Comment:
All the code after the flattening is identical with `tessellate_path`. Please extract to a common function instead of copy-pasting it.